smartsim/entity/strategies.py

```diff
@@ -40,31 +42,18 @@ def create_all_permutations(param_names, param_values):
     return all_permutations
 
 
-def step_values(param_names, param_values):
+def step_values(param_names: t.List[str], param_values: t.List[t.List[str]], _n_models: int = 0) -> t.List[t.Dict[str, str]]:
     permutations = []
     for p in zip(*param_values):
         permutations.append(dict(zip(param_names, p)))
     return permutations
 
 
-def random_permutations(param_names, param_values, n_models):
-    import random
-
+def random_permutations(param_names: t.List[str], param_values: t.List[t.List[str]], n_models: int = 0) -> t.List[t.Dict[str, str]]:
     # first, check if we've requested more values than possible.
-    perms = list(product(*param_values))
-    if n_models >= len(perms):
-        return create_all_permutations(param_names, param_values)
-    else:
-        permutations = []
-        permutation_strings = set()
-        while len(permutations) < n_models:
-            model_dict = dict(
-                zip(
-                    param_names,
-                    map(lambda x: x[random.randint(0, len(x) - 1)], param_values),
-                )
-            )
-            if str(model_dict) not in permutation_strings:
-                permutation_strings.add(str(model_dict))
-                permutations.append(model_dict)
-        return permutations
+    permutations = create_all_permutations(param_names, param_values)
+    
+    if n_models and n_models < len(permutations):
+        permutations = random.sample(permutations, n_models)
+    
+    return permutations
```

Comment:
Flagging this to come back and review more carefully -- big change to the implementation

Reply:
From MattD:
looks like that should be roughly equivalent, if not 1-to-1 with the current lib call. We don’t do any kind of setting a seed for random any where in this file, right? The only thing that would make me hold off is if this needs to be “reproducibly random”, but iirc we don’t do have that constraint for the the random perms strategy

Reply:
I think that this is ok, but let's document it somewhere. A comment here in the file would be sufficient

Reply:
pulled this refactor out into a separate PR.